docs/ff-concepts/navigation-routing/passing-data.md

```diff
@@ -0,0 +1,162 @@
+<p></p>
+
+When a page parameter is set to Required, it indicates that this parameter is
+mandatory when navigating to this page. Users must provide this value;
+otherwise, FlutterFlow will throw errors. However, if you are creating an
+optional parameter, please ensure this option is unchecked.
+
+Additionally, you can specify a default value in the Default Parameter Value
+field to safeguard against incoming values that are empty or null. This step is
+optional.
+
+![Page-Params.png](imgs/Page-Params.png)
+
+If you have created a **Required** Page Parameter and there is a Navigation Action
+already set on your previous page, FlutterFlow will throw errors because this
+required parameter has not yet been sent from the previous page. Let's fix that:
+
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/kp34JJipEW24hz0u5RsW?embed&show_copy_link=true"
+        title="Send Page Parameters"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+<figure>
+    
+  <figcaption class="centered-caption"></figcaption>
+</figure>
+
+:::info
+Passing data can only be tested in **Run** and **Test** Mode (it can not be tested in Preview Mode).
+:::
+ 
+## When to use Page Parameters?
+Page parameters are used to pass essential data between pages that is not
+persisted in the app’s global state but is necessary for specific
+functionalities or displays on the subsequent page. Here’s a
+breakdown of typical uses:
+
+- **Contextual Data:** Information that defines the context of the new page, such
+as
+identifiers for items or entities that the page must display. This could include
+identifiers for transactions, specific products, or user profiles that were
+selected on the previous page.
+
+- **Configuration Options:** Settings or options chosen by the user that affect
+how
+the next page functions or appears. For example, filter or sort preferences
+selected on a list page that need to be applied on a subsequent results page.
+
+- **Operational Parameters:** Values needed for calculations or logic on the next
+page
+that are generated through user activities on the current page. These could be
+values like quantities, dates, or configuration details necessary to perform
+operations or initiate processes on the next page.
+
+Page parameters are thus essential for maintaining a seamless user experience,
+enabling the new page to function as intended based on the specific needs and
+inputs from a previous interaction.
+
+## What you can pass
```

Comment:
Maybe rename it to "Allowed Data Types" or something similar?

Reply:
Ok, updated.